src/main/java/com/scrive/pdftools/tools/flattener/FlattenerSpec.java

```diff
@@ -0,0 +1,32 @@
+package com.scrive.pdftools.tools.flattener;
+
+import com.scrive.pdftools.tools.utils.FileInput;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class FlattenerSpec {
+    final private byte[] fileContent;
+    final private String documentNumberText;
+
+    public FlattenerSpec(byte[] fileContent, String documentNumberText) {
+        this.fileContent = fileContent;
+        this.documentNumberText = documentNumberText;
+    }
+
+    static public FlattenerSpec fromJson(final JSONObject obj) throws IOException {
+        final JSONObject mainFileInputObject = obj.getJSONObject("mainFileInput");
```

Comment:
also it is not necessary to mark parameters (and often even local variables) with `final`

Reply:
Fixed.